### PR TITLE
Review and Merge PR

### PR DESCRIPTION
**Pull request by KrishnaAlagiri**
## Problem Statements Added
- **December 15** - Pascal's triangle
- **December 16** - Find Temperature Difference Between Two Cities
## Other Changes
- Added **Header**
- Added **[Back to top]** at the bottom of the page

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
-# A December of Algorithms
+<div align="left">
```
Comment (akashsara) [inline README.md]: Good idea.

File: README.md [hunk 2]
```diff
@@ -279,6 +284,81 @@ We have a small collection of algorithms, one for every day of the month. Scroll
   - [More on the Caesar Cipher](http://practicalcryptography.com/ciphers/caesar-cipher)
 
 
+### **December 15 - Pascal's triangle**
```
Comment (akashsara) [inline README.md]: Use Title Case. Same for the index too.

File: README.md [hunk 3]
```diff
@@ -279,6 +284,81 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+- **Problem**
```
Comment (akashsara) [inline README.md]: Include this line at the start of the problem:
"Pascal's Triangle is a triangle(represented as a 2-D array) in which the ends of the rows are 1. Each of the other values is equal to the sum of the two nearest numbers in the row above."

File: README.md [hunk 4]
```diff
@@ -279,6 +284,81 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+rows = 3
+Sample output:
+1       --> row 0
+11      --> row 1
+121     --> row 2
+1331    --> row 3
+Polynomial: (1x^3y^0)+(3x^2y^1)+(3x^1y^2)+(1x^0y^3)
+```
+- **Uses**
+  -  The numbers in Pascal's triangle have another name: Binomial coefficient, and they show up a bunch. They're generally written like `(nx)(nx)` where `n` is the row and `x` is the position (starting from 0, not one.)
```
Comment (akashsara) [inline README.md]: Is it supposed to be (nx)(nx)? Just double check this part alone.

File: README.md [hunk 5]
```diff
@@ -279,6 +284,81 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+
+  - There is a formula for finding them, but for smaller numbers it's about as easy, or easier, to use the triangle as it is the formula.
+- **Application**
```
Comment (akashsara) [inline README.md]: Each of the applications are at the same indent level as the Application heading itself. Add a little indentation. 

image

File: README.md [hunk 6]
```diff
@@ -279,6 +284,81 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+- **Binomial Expansion**
```
Comment (akashsara) [inline README.md]: This is actually pretty cool. Didn't know this. Kudos to whoever made this problem.

File: README.md [hunk 7]
```diff
@@ -279,6 +284,81 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+  - It’s getting harder to turn around in tech without bumping into some reference to APIs, or application programming interfaces. In the simplest terms, APIs are sets of requirements that govern how one application can talk to another.
+  - OpenWeatherMap is a popular service that provides satellite images, Weather data, Historical data and Soil temperature and moisture.
+  - The problem is to find the current temperature difference between `London, GB` and `Chennai, Tamil Nadu, India` using the OpenWeatherMap API.
+  - `Note:` OpenWeatherMap APIs need a valid API key to allow responses, Packages won't work if you don't provide one. You can signup for a free API key on the OpenWeatherMap website.
+- **Sample Output**
+  ```
+  Taken at 20:35 (IST), 04-12-2018
+  Current Temperature difference between Chennai,TN and London,GB is 19°C
+  ```
+- **Resources:**
```
Comment (akashsara) [inline README.md]: Add: 
[How to retrieve weather data for a city](https://openweathermap.org/current#name)

File: README.md [hunk 8]
```diff
@@ -279,6 +284,81 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+    1   5   10   10   5   1
+    ```
+    - At a glance we can see the number of possibilities for 0, 1, 2, 3, 4, and 5 successes in 5 trials. To find the probability of 2 sixes, we just have to multiply (1/6)^2(5/6)^4 by 10.
+- **Resources:**
+  - [More about pascals triangle and how they work](https://www.youtube.com/watch?v=YUqHdxxdbyM)
+
+
+### **December 16 - Find Temperature Difference Between Two Cities**
+- **Problem**
+  - It’s getting harder to turn around in tech without bumping into some reference to APIs, or application programming interfaces. In the simplest terms, APIs are sets of requirements that govern how one application can talk to another.
```
Comment (akashsara) [inline README.md]: - It’s getting harder to turn around in tech without bumping into some reference to APIs, or application programming interfaces. In the simplest terms, applications talk to each other via APIs. 

- Usually, when we use an API to retrieve data (known as an API Call), we receive a JSON file containing data. To access this data, we will need to parse the JSON file. Depending on the language you are using, there are different ways to parse it. Just Google how to parse a JSON file in your language.

File: README.md [hunk 9]
```diff
@@ -279,6 +284,81 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+  - OpenWeatherMap is a popular service that provides satellite images, Weather data, Historical data and Soil temperature and moisture.
+  - The problem is to find the current temperature difference between `London, GB` and `Chennai, Tamil Nadu, India` using the OpenWeatherMap API.
+  - `Note:` OpenWeatherMap APIs need a valid API key to allow responses, Packages won't work if you don't provide one. You can signup for a free API key on the OpenWeatherMap website.
```
Comment (akashsara) [inline README.md]: Add an extra note stating: 
"API keys are private and should not be made public. When committing your code, please make sure you replace your original API key with a dummy key. 

Alternatively, you can look into saving your API key as an Environment Variable. 

One final alternative is to simply store the API key in a text file and add that file to your .gitignore. When you need to access it, open the file using your program and retrieve the API key."

File: README.md [hunk 10]
```diff
@@ -279,6 +284,81 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+- **Sample Output**
+  ```
+  Taken at 20:35 (IST), 04-12-2018
+  Current Temperature difference between Chennai,TN and London,GB is 19°C
+  ```
+- **Resources:**
+  - [How to get your API key from OpenWeatherMap](https://openweathermap.org/appid)
+  - [Java package for use with openweathermap.org's API.](https://sourceforge.net/projects/openjavaweatherapi/)
```
Comment (akashsara) [inline README.md]: I actually think it'll be better if we don't include the packages. They simplify the process too much imo. But I'll let you all have the final say on that.

File: README.md (hunk 1 above)
Comment (KrishnaAlagiri) [reply README.md]: Thanks :)

File: README.md (hunk 6 above)
Comment (KrishnaAlagiri) [reply README.md]: It was Mahalakshmi 🥇

File: README.md (hunk 10 above)
Comment (KrishnaAlagiri) [reply README.md]: You're right, adding packages will make the job easier. I'll remove 'em immediately.

**Comment by KrishnaAlagiri** [comment]
###  Changes Made:
- Resolved Issues in **December 15** and **December 16**

File: README.md [hunk 11]
```diff
@@ -279,6 +284,86 @@ We have a small collection of algorithms, one for every day of the month. Scroll
...
+  - Pascal's Triangle is a triangle(represented as a 2-D array) in which the ends of the rows are 1. Each of the other values is equal to the sum of the two nearest numbers in the row above.
+  - The pascal’s triangle is given as:
+  ```
+	                    1
+                  1          1
+             1          2        1
+         1         3         3      1
+    1       4          6         4     1
+  ```
+ - Up To any number of rows. The task here is to generate a pascal’s triangle for n rows and using this find `(x+y)^n`
```
Comment (akashsara) [inline README.md]: Needs one more level of indentation.

Comment (KrishnaAlagiri) [reply README.md]: What am I supposed to intend here?

Comment (akashsara) [reply README.md]: The "Up to any number" line. It's on the same level as the headings.

Comment (KrishnaAlagiri) [reply README.md]: Sorry, just noticed.

**Comment by KrishnaAlagiri** [comment]
#### All Issues Resolved

**Pull request merged**